src/unittest/test_irr_gltf_mesh_loader.cpp

```diff
@@ -20,10 +19,16 @@ const auto gamespec = findSubgame("devtest");
 if (!gamespec.isValid())
 	SKIP();
 
-irr::scene::CSceneManager smgr(nullptr, nullptr, nullptr);
-const auto loadMesh = [&smgr](const irr::io::path& filepath) {
-	irr::io::CReadFile file(filepath);
-	return smgr.getMesh(&file);
+irr::SIrrlichtCreationParameters p;
+p.DriverType = video::EDT_NULL;
+auto *driver = irr::createDeviceEx(p);
+REQUIRE(driver);
+
+auto *smgr = driver->getSceneManager();
+const auto loadMesh = [&] (const io::path& filepath) {
+	irr_ptr<io::IReadFile> file(driver->getFileSystem()->createAndOpenFile(filepath));
+	REQUIRE(file);
+	return smgr->getMesh(file.get());
```

Comment:
note: irrlicht caches meshes internally so this is not an irr_ptr

Reply:
why isnt this just a regular function or whatever? What's the point here?

Reply:
I have no idea what you are saying

Reply:
> Why is a lambda being used over a function?

IIRC because this is in a `TEST_CASE("gltf") {`, and since it is uses local variables (`driver`), it can't be moved out of it.

Similar to a namespace I prefer to not indent the `TEST_CASE` because I don't like it if the entire meat of the test has +1 indentation level.

Reply:
@appgurueu Can't functions be in blocks now?

Reply:
This "block" will be something like a lambda or function body itself based on how the `TEST_CASE` macro is defined.

A function which binds to local variables is a lambda and there is no problem with that. You could instead use a function and pass the variables instead, but eh.